### PR TITLE
NEW Allow state to be passed back on grid field actions

### PR DESCRIPTION
**Pull request by ScopeyNZ**
This PR is a benign addition to GridField actions to pass through an "action-state" data attribute (if available) into the request that performs an action. This is targetting 1.3 as it's part of a fix to prevent cramming the session with unnessecary data which doesn't scale well.

See related PR silverstripe/silverstripe-framework#8627

### CODE REVIEW

**Comment by ScopeyNZ** [comment]
I've just noticed there's an issue with `GridFieldFilterHeader`. Looking into it 😪

**Comment by bergice** [comment]
Tests need to be fixed up.

**Comment by ScopeyNZ** [comment]
Looks like PECL was having some problems yesterday. I've restarted the builds

File: client/src/legacy/GridField.js [hunk 1]
```diff
@@ -366,21 +366,33 @@ $.entwine('ss', function($) {
...
+      var data = [
         {
-          data: [{
-            name: this.attr('name'),
-            value: this.val(),
-            filter: filterState,
-            triggerChange
-          }],
+          name: this.attr('name'),
+          state: 'abc',
```
Comment (maxime-rainville) [inline client/src/legacy/GridField.js]: This looks like a debug attribute.

Comment (ScopeyNZ) [reply client/src/legacy/GridField.js]: Yep. I'll remove this.

Comment (ScopeyNZ) [reply client/src/legacy/GridField.js]: Removed.

**Comment by ScopeyNZ** [comment]
I've just rebased. I don't expect that a rebase will fix the failing builds that aren't related. 

I don't have time to address the failing Behat tests within this PR.

**Comment by ScopeyNZ** [comment]
I think I found an issue so please don't merge yet.

**Comment by maxime-rainville** [comment]
@guy Have you found/resolve your this new mysterious issue?

**Comment by ScopeyNZ** [comment]
Yeah. The issue was the same that is fixed by #765 . Sorry we're good to merge given that's fixed 🙂

**Comment by maxime-rainville** [comment]
Merging everything before anyone has a chance to say otherwise.

**Comment by robbieaverill** [comment]
But....

**Comment by robbieaverill** [comment]
Jokes

**Pull request merged**